### PR TITLE
Add base classes

### PR DESCRIPTION
**Pull request by lcc**
Add base classes and the KNoT state machine manager.

### CODE REVIEW

File: knot-thing/src/main/java/com/cesar/knot_sdk/knot_state_machine/KNoTStateMachine.kt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    fun enterState() {
+        state.enter()
+    }
+
+    fun getUUUID() = sharedPref.getString(
+        UUID_PREF,
+        PREF_MISSING
+    )
+
```
Comment (alla3) [inline knot-thing/src/main/java/com/cesar/knot_sdk/knot_state_machine/KNoTStateMachine.kt:52]: Remove a blank line.

File: knot-thing/src/main/java/com/cesar/knot_sdk/knot_state_machine/states/base_classes/State.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package com.cesar.knot_sdk.knot_state_machine.states.base_classes
+
+abstract class State {
+
+
```
Comment (alla3) [inline knot-thing/src/main/java/com/cesar/knot_sdk/knot_state_machine/states/base_classes/State.kt]: Remove a blank line.

File: knot-thing/src/main/java/com/cesar/knot_sdk/knot_state_machine/states/base_classes/State.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    enum class ReadyStates{
```
Comment (alla3) [inline knot-thing/src/main/java/com/cesar/knot_sdk/knot_state_machine/states/base_classes/State.kt]: Add a space before '{'.

File: knot-thing/src/main/java/com/cesar/knot_sdk/knot_state_machine/states/base_classes/State.kt (hunk 2 above)
Comment (lcc) [reply knot-thing/src/main/java/com/cesar/knot_sdk/knot_state_machine/states/base_classes/State.kt]: Done.

File: knot-thing/src/main/java/com/cesar/knot_sdk/knot_state_machine/states/base_classes/State.kt (hunk 3 above)
Comment (lcc) [reply knot-thing/src/main/java/com/cesar/knot_sdk/knot_state_machine/states/base_classes/State.kt]: Done.

File: knot-thing/src/main/java/com/cesar/knot_sdk/knot_state_machine/KNoTStateMachine.kt (hunk 1 above)
Comment (lcc) [reply knot-thing/src/main/java/com/cesar/knot_sdk/knot_state_machine/KNoTStateMachine.kt:52]: Done.

**Review by alla3 — APPROVED** [review]

**Pull request merged**